### ISSUE TITLE
Running on macOS, it crashes after about 2 minutes or so if it doesn't show up at the top.

### ISSUE THREAD

**Issue by D10NGYANG** [JetBrains/compose-jb#940]
LOG：
```
> Task :run
2021-07-27 14:42:50.715 java[7789:583307] TSM AdjustCapsLockLEDForKeyTransitionHandling - _ISSetPhysicalKeyboardCapsLockLED Inhibit
2021-07-27 14:43:35.763 java[7789:583307] +[AWTView keyboardInputSourceChanged:]: unrecognized selector sent to class 0x10b52d5e8
2021-07-27 14:43:35.764 java[7789:583307] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '+[AWTView keyboardInputSourceChanged:]: unrecognized selector sent to class 0x10b52d5e8'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff206d287b __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff2040ad92 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff207552e5 __CFExceptionProem + 0
	3   CoreFoundation                      0x00007fff2063a90b ___forwarding___ + 1448
	4   CoreFoundation                      0x00007fff2063a2d8 _CF_forwarding_prep_0 + 120
	5   CoreFoundation                      0x00007fff2064e10d __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	6   CoreFoundation                      0x00007fff206e9b6c ___CFXRegistrationPost_block_invoke + 49
	7   CoreFoundation                      0x00007fff206e9aea _CFXRegistrationPost + 454
	8   CoreFoundation                      0x00007fff2061f38e _CFXNotificationPost + 736
	9   Foundation                          0x00007fff2138ddc8 -[NSNotificationCenter postNotificationName:object:userInfo:] + 59
	10  AppKit                              0x00007fff2320065b __NSSelectedKeyboardInputSourcesChangedNotification + 58
	11  CoreFoundation                      0x00007fff2064e10d __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	12  CoreFoundation                      0x00007fff206e9b6c ___CFXRegistrationPost_block_invoke + 49
	13  libdispatch.dylib                   0x00007fff203b4623 _dispatch_call_block_and_release + 12
	14  libdispatch.dylib                   0x00007fff203b5806 _dispatch_client_callout + 8
	15  libdispatch.dylib                   0x00007fff203c1b4f _dispatch_main_queue_callback_4CF + 940
	16  CoreFoundation                      0x00007fff206955e8 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	17  CoreFoundation                      0x00007fff206577d2 __CFRunLoopRun + 2755
	18  CoreFoundation                      0x00007fff2065664c CFRunLoopRunSpecific + 563
	19  HIToolbox                           0x00007fff28894ab3 RunCurrentEventLoopInMode + 292
	20  HIToolbox                           0x00007fff28894815 ReceiveNextEventCommon + 587
	21  HIToolbox                           0x00007fff288945b3 _BlockUntilNextEventMatchingListInModeWithFilter + 70
	22  AppKit                              0x00007fff22e576f2 _DPSNextEvent + 864
	23  AppKit                              0x00007fff22e55ec5 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1364
	24  libosxapp.dylib                     0x000000010b56db3a -[NSApplicationAWT nextEventMatchingMask:untilDate:inMode:dequeue:] + 121
	25  AppKit                              0x00007fff22e48239 -[NSApplication run] + 586
	26  libosxapp.dylib                     0x000000010b56d97f +[NSApplicationAWT runAWTLoopWithApp:] + 219
	27  libawt_lwawt.dylib                  0x000000010b4f8588 +[AWTStarter starter:headless:] + 809
	28  JavaNativeFoundation                0x00007fff6eb2177f +[JNFRunLoop _performCopiedBlock:] + 15
	29  Foundation                          0x00007fff21404bc1 __NSThreadPerformPerform + 204
	30  CoreFoundation                      0x00007fff2065897c __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	31  CoreFoundation                      0x00007fff206588e4 __CFRunLoopDoSource0 + 180
	32  CoreFoundation                      0x00007fff20658664 __CFRunLoopDoSources0 + 242
	33  CoreFoundation                      0x00007fff2065708c __CFRunLoopRun + 893
	34  CoreFoundation                      0x00007fff2065664c CFRunLoopRunSpecific + 563
	35  libjli.dylib                        0x0000000108a035b5 CreateExecutionEnvironment + 398
	36  libjli.dylib                        0x00000001089ff81f JLI_Launch + 1315
	37  java                                0x00000001089f12cc main + 364
	38  libdyld.dylib                       0x00007fff2057af5d start + 1
)
libc++abi: terminating with uncaught exception of type NSException

> Task :run FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk-11.0.11.jdk/Contents/Home/bin/java'' finished with non-zero exit value 134
```

macOS 11.4
jdk 11.0.11

**Comment by akurasov** [comment]
Could you please share a reproducer? Is it something like "wait 2 minutes", then start Composition?

**Comment by D10NGYANG** [comment]
> Could you please share a reproducer? Is it something like "wait 2 minutes", then start Composition?

Here's my code repository 
https://gitee.com/d10ng/DesktopTest
You can download it and test it
There is no time delay operation in it

**Comment by m-sasha** [comment]
The crash happens whenever there's an "active" TextField, and the input source (input language) changes. It reproduces every time for me:

```
2022-05-09 15:25:08.674 java[5032:117330] +[AWTView keyboardInputSourceChanged:]: unrecognized selector sent to class 0x10bc023b8
2022-05-09 15:25:08.677 java[5032:117330] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '+[AWTView keyboardInputSourceChanged:]: unrecognized selector sent to class 0x10bc023b8'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007ff81ff181e3 __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007ff81fc78c13 objc_exception_throw + 48
	2   CoreFoundation                      0x00007ff81ff9ce93 __CFExceptionProem + 0
	3   CoreFoundation                      0x00007ff81fe7edf5 ___forwarding___ + 1356
	4   CoreFoundation                      0x00007ff81fe7e818 _CF_forwarding_prep_0 + 120
	5   CoreFoundation                      0x00007ff81fe9203c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	6   CoreFoundation                      0x00007ff81ff2f592 ___CFXRegistrationPost_block_invoke + 49
	7   CoreFoundation                      0x00007ff81ff2f510 _CFXRegistrationPost + 496
	8   CoreFoundation                      0x00007ff81fe63ca8 _CFXNotificationPost + 735
	9   Foundation                          0x00007ff820c9890e -[NSNotificationCenter postNotificationName:object:userInfo:] + 82
	10  AppKit                              0x00007ff822c3b2f7 __NSSelectedKeyboardInputSourcesChangedNotification + 58
	11  CoreFoundation                      0x00007ff81fe9203c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	12  CoreFoundation                      0x00007ff81ff2f592 ___CFXRegistrationPost_block_invoke + 49
	13  libdispatch.dylib                   0x00007ff81fc1a0cc _dispatch_call_block_and_release + 12
	14  libdispatch.dylib                   0x00007ff81fc1b317 _dispatch_client_callout + 8
	15  libdispatch.dylib                   0x00007ff81fc27c78 _dispatch_main_queue_drain + 943
	16  libdispatch.dylib                   0x00007ff81fc278bb _dispatch_main_queue_callback_4CF + 31
	17  CoreFoundation                      0x00007ff81feda9c7 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	18  CoreFoundation                      0x00007ff81fe9b93f __CFRunLoopRun + 2771
	19  CoreFoundation                      0x00007ff81fe9a7ac CFRunLoopRunSpecific + 562
	20  HIToolbox                           0x00007ff828b21ce6 RunCurrentEventLoopInMode + 292
	21  HIToolbox                           0x00007ff828b21913 ReceiveNextEventCommon + 283
	22  HIToolbox                           0x00007ff828b217e5 _BlockUntilNextEventMatchingListInModeWithFilter + 70
	23  AppKit                              0x00007ff8228c153d _DPSNextEvent + 927
	24  AppKit                              0x00007ff8228bfbfa -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1394
	25  libosxapp.dylib                     0x000000010b93b6fe -[NSApplicationAWT nextEventMatchingMask:untilDate:inMode:dequeue:] + 121
	26  AppKit                              0x00007ff8228b22a9 -[NSApplication run] + 586
	27  libosxapp.dylib                     0x000000010b93b543 +[NSApplicationAWT runAWTLoopWithApp:] + 263
	28  libawt_lwawt.dylib                  0x000000010bbcaa38 +[AWTStarter starter:headless:] + 496
	29  libosxapp.dylib                     0x000000010b93d059 +[ThreadUtilities invokeBlockCopy:] + 15
	30  Foundation                          0x00007ff820d10857 __NSThreadPerformPerform + 179
	31  CoreFoundation                      0x00007ff81fe9caeb __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	32  CoreFoundation                      0x00007ff81fe9ca53 __CFRunLoopDoSource0 + 180
	33  CoreFoundation                      0x00007ff81fe9c7cd __CFRunLoopDoSources0 + 242
	34  CoreFoundation                      0x00007ff81fe9b1e8 __CFRunLoopRun + 892
	35  CoreFoundation                      0x00007ff81fe9a7ac CFRunLoopRunSpecific + 562
	36  libjli.dylib                        0x000000010b1325b5 CreateExecutionEnvironment + 398
	37  libjli.dylib                        0x000000010b12e81f JLI_Launch + 1315
	38  java                                0x000000010b00d2cc main + 364
	39  dyld                                0x000000011a4da51e start + 462
)
libc++abi: terminating with uncaught exception of type NSException
```

**Comment by igordmn** [comment]
Changing layouts between English, Russian and Chinese works fine on my configuration:
macOs 11.6.1, OpenJDK 11.0.13, Compose 1.1.0.

```
import androidx.compose.desktop.ui.tooling.preview.Preview
import androidx.compose.material.Button
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.material.TextField
import androidx.compose.runtime.*
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.window.*
import androidx.compose.ui.window.MenuBar
import java.awt.event.MouseEvent
import java.awt.event.MouseMotionListener
import javax.swing.JMenuBar
import javax.swing.UIManager

@Composable
@Preview
fun App() {
    var text by remember { mutableStateOf("Hello, World!") }

    MaterialTheme {
        Button(onClick = {
            text = "Hello, Desktop!"
        }) {
            Text(text)
        }
    }
}

fun main() = application {
    var f by remember { mutableStateOf("") }
    Window(onCloseRequest = ::exitApplication) {
        TextField(f, {f = it})
    }
}
```

Could you try with a newer JDK?

**Comment by m-sasha** [comment]
It crashes for me on Oracle OpenJDK 11.0.12 (Compose 1.1.1), but does not crash on Oracle OpenJDK 11.0.15

**Comment by igordmn** [comment]
When I downgraded to OpenJDK 11.0.12, I caught this crash. After looking at JDK issues, I found [this](https://bugs.openjdk.java.net/browse/JDK-8257242?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel).

We can't do workaround in Compose for the older JDK, so recommendation is to use the latest patch of JDK.

I added info about this issue [here](https://github.com/JetBrains/compose-jb/blob/master/README.md)
